src/website/layout.gleam

```diff
@@ -0,0 +1,292 @@
+import gleam/dict
+import gleam/int
+import gleam/list
+import gleam/option.{type Option}
+import gleam/result
+import gleam/time/calendar
+import gleam/time/timestamp
+import jot
+import lustre/attribute.{attribute as attr, class} as attr
+import lustre/element.{type Element}
+import lustre/element/html
+import website/site
+
+pub type ContentLink {
+  ContentLink(title: String, href: String, children: List(ContentLink))
+}
+
+pub fn table_of_contents_from_djot(document: jot.Document) -> List(ContentLink) {
```

Comment:
Is this used anywhere?

Reply:
It is not currently, but it will be as soon as you use djot to write a guide like this.